### PR TITLE
41-42-service-status

### PR DESCRIPTION
**Pull request by 1AhmedYasser**
Task #41
Task #42

### CODE REVIEW

File: DSL/Liquibase/changelog/1678307467-add-ruuter-types.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- liquibase formatted sql
+-- changeset ahmedyasser:1678307467
+ALTER TABLE services 
+ADD COLUMN ruuter_type TEXT DEFAULT 'GET';
```
Comment (trevorling) [inline DSL/Liquibase/changelog/1678307467-add-ruuter-types.sql:4]: Are you sure that you want to allow `text` as a type?
I'd like to use an enum here.
Then you are able to use only the values which are defined in the enum. 
Now you are able to set the `ruuter_type` to `cat` for example.

File: DSL/Resql/services/get-service.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+SELECT name FROM services
+WHERE id = :id;
```
Comment (trevorling) [inline DSL/Resql/services/get-service.sql:2]: I'd rename the file:
`get-service-name-by-id.sql`
What do you think?

File: DSL/Ruuter/POST/services/status.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  call: http.post
+  args:
+    url: http://node_server:3003/file/move
+    body:
+      current_path: '/Ruuter/${ruuter_type}/services/inactive/${name_res.response.body[0].name}.yml'
+      new_path: '/Ruuter/${ruuter_type}/services/${name_res.response.body[0].name}.yml'
+  result: results  
+  next: check_result
+
+in_activate_service:
```
Comment (trevorling) [inline DSL/Ruuter/POST/services/status.yml:49]: `in_activate_service` -> `deactivate_service`
In my opinion, it's more descriptive.

File: README.md [hunk 4]
```diff
@@ -38,3 +38,5 @@ This repo will primarily contain:
 
 - For setting up the users database initially, run
   `docker run --platform linux/amd64 --network=bykstack riaee/byk-users-db:liquibase20220615 --url=jdbc:postgresql://database:5432/users_db --username=byk --password=01234 --changelog-file=./master.yml update`
+- Run migrations added in this repository by running the helper script `./migrate.sh`
+- When creating new migrations, use the helper `./create-migration.sh name-of-migration` which will create a timestamped file in the correct directory and add the required headers
```
Comment (trevorling) [inline README.md:42]: Thanks for the update! :)

**Review by trevorling — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (1AhmedYasser) [reply README.md:42]: Your Mostly Welcome 👍

**Review by baha-a — APPROVED** [review]

File: DSL/Liquibase/changelog/1678307467-add-ruuter-types.sql (hunk 1 above)
Comment (trevorling) [reply DSL/Liquibase/changelog/1678307467-add-ruuter-types.sql:4]:
```sql
CREATE TYPE ruuter_request_type AS ENUM ('GET', 'POST');

ALTER TABLE services 
ADD COLUMN ruuter_type ruuter_request_type DEFAULT 'GET';

File: DSL/Resql/services/add.sql [hunk 5]
```diff
@@ -1,2 +1,2 @@
-INSERT INTO services (name, description)
-VALUES (:name, :description);
+INSERT INTO services (name, description,ruuter_type)
+VALUES (:name, :description,:type);
```
Comment (trevorling) [inline DSL/Resql/services/add.sql:2]:
```sql
INSERT INTO services (name, description,ruuter_type)
VALUES (:name, :description,:ruuter_type::ruuter_request_type);

File: DSL/Ruuter/POST/services/add.yml [hunk 6]
```diff
@@ -15,14 +16,52 @@ service_add:
   args:
     url: http://resql:8082/add
     body:
-      name: ${incoming.body.name}
-      description: ${incoming.body.description}
+      name: ${name}
+      description: ${description}
+      type: ${type}
```
Comment (trevorling) [inline DSL/Ruuter/POST/services/add.yml:21]: `type` -> `ruuter_type`
```
ruuter_type: ${type}

File: DSL/Ruuter/POST/services/add.yml [hunk 7]
```diff
@@ -1,12 +1,13 @@
-check_for_body:
-  switch:
-    - condition: ${incoming.body == null}
-      next: return_incorrect_request
-  next: check_for_required_parameters
+extract_request_data:
+  assign:
+    name: ${incoming.body.name}
+    description: ${incoming.body.description}
+    type: ${incoming.body.type}
```
Comment (trevorling) [inline DSL/Ruuter/POST/services/add.yml:5]: If possible then please do:  `.toUpperCase()`

File: DSL/Ruuter/POST/services/add.yml [hunk 8]
```diff
@@ -15,14 +16,52 @@ service_add:
...
   result: createdService
+  next: check_for_type
 
-return_value:
-  status: 201
-  wrapper: FALSE
-  return: ""
+check_for_type:
+  switch:
+    - condition: ${type === 'GET'}
```
Comment (trevorling) [inline DSL/Ruuter/POST/services/add.yml:27]:
```
${type === 'GET' || type === 'get'}

File: DSL/Ruuter/POST/services/add.yml [hunk 9]
```diff
@@ -18,13 +18,13 @@ service_add:
     body:
       name: ${name}
       description: ${description}
-      type: ${type}
+      ruuter_type: ${type.toUpperCase()}
```
Comment (trevorling) [inline DSL/Ruuter/POST/services/add.yml:21]: Please use this `.toUpperCase()` in line 5.
Then you don't need to do it in multiple places.

**Review by trevorling — CHANGES_REQUESTED** [review]

**Review by trevorling — APPROVED** [review]

**Pull request merged**